### PR TITLE
service/header, cmd: Inject head by trusted header hash

### PR DESCRIPTION
**Pull request by renaynay**
TODO: 

- [ ] resolve broken TestFullLifecycle

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -58,11 +63,17 @@ func Start(repoName string, tp node.Type) *cobra.Command {
...
+			if headHash != "" {
+				opts = append(opts, node.WithHead(tmbytes.HexBytes(headHash)))
+			}
+
 			return start(cmd, tp, repo, opts...)
 		},
 	}
 
 	cmd.Flags().String(cfgAddress, "", "Indicates node to connect to the given remote core node")
+	cmd.Flags().String(headHash, "", "known header hash with which to initialize the node")
```
Comment (liamsi) [inline cmd/start.go:76]:
```suggestion
	cmd.Flags().String(headHash, "", "subjectively trusted header hash with which to initialize the node")
```

File: node/config.go [hunk 2]
```diff
@@ -20,6 +22,13 @@ type Config struct {
 	Core core.Config
 	P2P  p2p.Config
 	RPC  rpc.Config
+	// hash of the ExtendedHeader of the first block,
+	// this is necessary to begin the header.Service as
+	// it requires a header to already be injected in the store
+	// before it can begin syncing, for verification purposes.
+	// TODO @renaynay + @Wondertan: this is a solution for Devnet
+	// and should be modified in the future.
```
Comment (liamsi) [inline node/config.go:30]: Are we tracking the shortcuts we are taking for devnet somewhere? 
There is also: 
- https://github.com/celestiaorg/celestia-core/pull/582
- https://github.com/celestiaorg/lazyledger-app/pull/149

I feel there was another one in node?

**Review by liamsi — COMMENTED** [review]
Will re-review this with a fresh mind tomorrow. But I think this looks good.

**Comment by Wondertan** [comment]
Unfortunately, this attempt we did with Rene is wrong and there are many issues here. 
Closing in favor of: https://github.com/celestiaorg/celestia-node/pull/220

**Pull request closed without merging**